src/VisualStudio/Core/Def/Implementation/AbstractVsTextViewFilter.cs

```diff
@@ -258,6 +258,10 @@ internal static int GetPairExtentsWorker(ITextView textView, IBraceMatchingServi
                                 }
                             }
                         }
+                        else
+                        {
+                            return VSConstants.E_FAIL;
```

Comment:
i think we shoudl just return S_OK on lines 239 and 256.  tehn this function can return S_FALSE instead.  E_FAIL is equivalent to an exception and is basically us saying "we failed entirely on this function call".  S_FALSE is more like "hey... we proceeded just fine, we just didn't have any data to give you.

Reply:
`S_FALSE` won't work because the VS command handler uses [`ErrorHandler.Succeeded`](https://dev.azure.com/devdiv/DevDiv/_git/VS?path=/src/help/src/env/Shell/ErrorHandler.cs&version=GBmain&line=22&lineEnd=31&lineStartColumn=9&lineEndColumn=10&lineStyle=plain&_a=contents) to check if the text view filter did handle the brace matching command correctly: https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=/src/Editor/VisualStudio/Impl/ViewAdapter/IOleCommandTarget.cs&version=GBmain&line=3804&lineEnd=3826&lineStartColumn=25&lineEndColumn=26&lineStyle=plain&_a=contents

Reply:
Looking at that code, it already seems to understand that S_FALSE is a thing:

![image](https://user-images.githubusercontent.com/4564579/187795861-b245c9ef-1cd7-4712-927e-91e568c76871.png)

It just doesn't check that after making the second call.

Reply:
Yep, I should have specified that that was the exact if-statement that's preventing us from using `S_FALSE`.

Reply:
i guess my point is: we can change that 'if' on the platform side.  the code above clearly can handle S_FALSE as part of teh contract of GetPairExtents.  So it shoudl be trivial to update to check for that after the second call.

Reply:
Sounds reasonable. I'll update this PR to use `S_FALSE` instead then.